File: ArrayAndHashing/C++/ValidaAnagram.cpp
#include <map>
#include <vector>
#include <iostream>

using std::vector;
using std::map;
using std::string;

class Solution {
public:
    bool isAnagram(string& s, string& t) 
    {
        map<char, int> seen;

        if( s.size() != t.size())
            return false;

        for(int i = 0; i < s.size(); i++)
        {
            if( !seen.count(s[i]) )
            {
                seen[s[i]] = 1;
            }
            else
            {
                seen[s[i]] ++;
            }
        }

        for( int j = 0; j < t.size(); j++)
        {
            if( !seen.count( t[j]) )
            {
                return false;
            }
            else
            {
                seen[ t[j] ] -- ;
            }
        }

        for(int k = 0 ; k < seen.size(); k++)
        {
            if( seen[k] != 0 )
            {
                return false;
            }
        }
   
        return true;
    }
};

int main()
{
    string s = "anagram";
    string t = "nagaram";

    Solution sol;

    std::cout << sol.isAnagram(s, t);
}

File: ArrayAndHashing/python/ValidaAnagram.py
class Solution:
    def isAnagram(self, s: str, t:str) -> bool:
        if len(s)!= len(t):
            return False
        
        hash_map = {}
        
        for i in s:
            if i not in hash_map:
                hash_map[i] = 1
            else:
                hash_map[i] += 1

        for j in t:
            if j in hash_map:
                hash_map[j] -= 1

        for k in hash_map:
            if hash_map[k] != 0:
                return False

        return True

a = Solution()

s = "a"
t = "nagaram"
print(a.isAnagram(s, t))

s = {'a':0, 'b':5, 'c':6}

s.get('a',3)

